engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -157,6 +164,38 @@ private Either<Rejection, DeployedProcess> validateElementExists(final DeployedP
     }
     return Either.right(process);
   }
+  private Either<Rejection, DeployedProcess> validateElementsNotInsideMultiInstance(final DeployedProcess process,
+      final List<ProcessInstanceModificationActivateInstructionValue> activateInstructions) {
+    final Set<String> elementsInsideMultiInstance =
+        activateInstructions.stream()
+            .map(ProcessInstanceModificationActivateInstructionValue::getElementId)
+            .filter(elementId -> isInsideMultiInstanceBody(process, BufferUtil.wrapString(elementId)))
+            .collect(Collectors.toSet());
+    if (!elementsInsideMultiInstance.isEmpty()) {
+      final String reason =
+          String.format(
+              ERROR_MESSAGE_TARGET_ELEMENT_IN_MULTI_INSTANCE_BODY,
+              BufferUtil.bufferAsString(process.getBpmnProcessId()),
+              String.join("', '", elementsInsideMultiInstance));
+      return Either.left(new Rejection(RejectionType.INVALID_ARGUMENT, reason));
+    }
+    return Either.right(process);
```

Comment:
🔧 It might be slightly more readable to invert this if. 

The reasoning would be that this method is validating something. If there are no problems, it doesn't have to do anything and can early return `Either.right(process)`. Otherwise, it needs to create a rejection.

Note that originally the intention was to write all the validations in the same method and that the last return would mean that all is good and no rejections were encountered. But since you've switched to Either, this intention is no longer there.